src/IdealizedSimulations/neverworld_simulation.jl

```diff
@@ -0,0 +1,298 @@
+# Neverworld
+#
+# Ingredients:
+#
+#   * Zonally-periodic domain with continental shelves on all boundaries except Southern Ocean
+#       * longitude = (0, 60)
+#       * 2 configurations in latitude
+#           - Half basin: latitude = (-70, 0)
+#           - Full basin: latitude = (-70, 70)
+#       * z to -4000m
+#       * Southern Ocean channel from -60 to -40 (with a ridge to -2000 m)
+#
+#   * Zonally-homogeneous wind stress with mid-latitude jet and trade winds
+#
+#   * Buoyancy 
+#       * restoring hot at the equator and cold at the poles (parabola, cosine, smooth step function)
+#       * equator-pole buoyancy differential: 0.06 (α * g * 30 ≈ 0.06 with α=2e-4, g=9.81)
+#       * exponential initial vertical stratification with N² = 6e-5 and decay scale h = 1000 m
+#           - eg bᵢ = N² * h * exp(z / h)
+#
+#   * Quadratic bottom drag with drag_coefficient = 1e-3
+
+using CubicSplines
+
+using Oceananigans
+using Oceananigans.Units
+using Oceananigans.Grids: xnode, ynode
+using Oceananigans.Operators: Δzᶜᶜᶜ
+using Oceananigans.Coriolis: ActiveCellEnstrophyConservingScheme
+using Oceananigans.TurbulenceClosures: CATKEVerticalDiffusivity
+
+using ClimaOcean: CubicSplineFunction,
+                  u_bottom_drag, u_immersed_bottom_drag,
+                  v_bottom_drag, v_immersed_bottom_drag
+                  
+using ..VerticalGrids: stretched_vertical_cell_interfaces
+
+const c = Center()
+const f = Face()
+
+#####
+##### Geometry
+#####
+
+struct Point{T}
+    x :: T
+    y :: T
+end
+
+struct LineSegment{T}
+    p₁ :: Point{T}
+    p₂ :: Point{T}
+end
+
+struct Line{T}
+    p₁ :: Point{T}
+    p₂ :: Point{T}
+end
+
+distance(p₁::Point, p₂::Point) = sqrt((p₁.x - p₂.x)^2 + (p₁.y - p₂.y)^2)
+
+# https://stackoverflow.com/questions/849211/shortest-distance-between-a-point-and-a-line-segment
+function distance(point::Point, line::LineSegment)
+    x, y = point.x, point.y
+    x₁, y₁ = line.p₁.x, line.p₁.y
+    x₂, y₂ = line.p₂.x, line.p₂.y
+
+    # Line segment vector components
+    ℓˣ = x₂ - x₁
+    ℓʸ = y₂ - y₁
+
+    # Fractional increment to closest segment point
+    ϵ = ((x - x₁) * ℓˣ + (y - y₁) * ℓʸ) / (ℓˣ^2 + ℓʸ^2)
+    ϵ = clamp(ϵ, 0, 1)
+
+    # Closest segment point
+    x′ = x₁ + ϵ * ℓˣ
+    y′ = y₁ + ϵ * ℓʸ
+    
+    return sqrt((x′ - x)^2 + (y′ - y)^2)
```

Comment:
we can use the method `distance(p₁::Point, p₂::Point)` here? or delete that method?

```suggestion
    return distance(Point(x, y), Point(x′, y′))
```